src/main.py

```diff
@@ -1,8 +1,67 @@
-from fastapi import FastAPI
+import uuid
+
+from fastapi import FastAPI, HTTPException
+from src.schemas import ProjectDetails, Project
 
 app = FastAPI()
+projects: list[ProjectDetails] = []
+
+
+def get_project(project_id: uuid.UUID) -> ProjectDetails:
+    project = next((p for p in projects if p.project_id == project_id), None)
+    if project is None:
+        raise HTTPException(status_code=404, detail=f"ProjectDetails {project_id} not found")
+    return project
+
+
+def create_project_service(project: Project) -> ProjectDetails:
+    new_project = ProjectDetails(**project.model_dump())
+    projects.append(new_project)
+    return new_project
+
+
+def get_projects_service() -> list[ProjectDetails]:
+    return projects
+
+
+def get_project_details_service(project_id: uuid.UUID) -> ProjectDetails:
+    project = get_project(project_id)
+    return project
+
+
+def update_project_details_service(project_id: uuid.UUID, name: str, description: str | None) -> ProjectDetails:
+    project = get_project(project_id)
+    project.description = description
+    project.name = name
+    return project
+
+
+def delete_project_service(project_id: uuid.UUID) -> ProjectDetails:
+    project = get_project(project_id)
```

Comment:
move this out of delete function. otherwise you mix up logics and it will be harder to change something later